ui/src/testing/stubs/workspace-service-stub.ts

```diff
@@ -210,4 +211,21 @@ export class WorkspacesServiceStub {
       }, 0);
     });
   }
+
+  getNoteBookList(workspaceNamespace: string,
+      workspaceId: string, extraHttpRequestParams?: any): Observable<Array<FileDetail>> {
+    return new Observable<Array<FileDetail>>(observer => {
+      setTimeout(() => {
+        if (workspaceNamespace === WorkspaceStubVariables.DEFAULT_WORKSPACE_NS
+            && workspaceId === WorkspaceStubVariables.DEFAULT_WORKSPACE_ID) {
+          const fileDetailsList =
+              [{'name': 'FileDetails', 'path': 'gs://bucket/notebook/mockFile'}];
+          observer.next(fileDetailsList);
+        } else {
+          observer.next([]);
+        }
+        observer.complete();
+      }, 0);
```

Comment:
This `setTimeout(..., 0)` pattern doesn't make sense to me (I know you're just following the existing convention). @blrubenstein Do you know why we need these?

Reply:
This is a better simulation of a remote call. Calling `subscribe()` on an `Observable` results in synchronous callback execution, if the Observable already provided a value. In the XHR case, this Observable would only provide a value (call made to `observer.next()`) asynchronously.

You likely already know this, but for full context `setTimeout(..., 0)` is a standard pattern in JS which basically says "let the current execution stack fully resolve before executing this callback"; in this situation it means that any immediate subscription to this observable will be satisfied asynchronously. I made a demo to verify this behavior: https://goo.gl/xvRjBP

Reply:
Makes sense, but I'd probably pull this out into a function just to clarify. Perhaps something like:
```javascript
function simulateAsync(f) {
  setTimeout(f, 0);
}
```

Reply:
Well, it's not simulating async, it is making it async. You want it to be async here in order to simulate an XHR. I don't feel strongly, but my 2c would be to not bother obfuscating it: the method naming chosen will inevitably be ambiguous and a general developer would need to click in to understand exactly what it's doing (then they'll need to learn setTimeout anyways). It's also a pretty common pattern in JS that is worth understanding/learning. Note that this pattern is not only used in testing.